src/Events.ts

```diff
@@ -6,16 +6,44 @@ export interface EventDeclaration {
     [slotName: string]: Slot<any, any>
 }
 
-export function combineEvents<C extends EventDeclaration>(...args: C[]): { [P in keyof C]: C[P] } {
-    // TODO: throw if event buses have duplicate events
+export function combineEvents<
```

Comment:
I am not sure that I understand your comment.
The reason why I went back to an older version (with some improvements) of this `combineEvents` function is that the "new" implementation was expecting every provided event declarations to be of the same `C` type, which is certainly not the intention: the goal is to combine different sources of events.

Reply:
The author who made the change did write a test and thought is was working, but the subtlety is that the test was written like this:
```
          const combined = combineEvents(
              {
                  hello: slot<{ name: string }>()                                                                                                                                                                   
              },
              {
                  how: slot<{ mode: 'simple' | 'advanced'}>(),
                  are: slot<{ tense: number }>()
              },
              {
                  you: slot<{ reflective: boolean }>()
              }
          )

```
Where typescript correctly infers that `C` is the union of the three types: it understand that this type has been provided three times.

But this syntax would fail:
```
          const helloEvents = {
              hello: slot<{ name: string }>()
          }
          const howAreEvents = {
              how: slot<{ mode: 'simple' | 'advanced'}>(),
              are: slot<{ tense: number }>()
          }
          const youEvents = {
              you: slot<{ reflective: boolean }>()
          }
          const combined = combineEvents(
              helloEvents,
              howAreEvents,
              youEvents
          )
```
Because then typescript sees 3 different types and can't decide what `C` is with the erroneous implementation.